### ISSUE TITLE
Add a "whitelist" function

### ISSUE THREAD

**Issue by RaymondSchnyder** [diantahoc/chan-archiver#8]
Hey,

thx for your nice Software, but i have an idea.
can you add a filter which only Downloads threads which are on the whitelist?
Example:
Filter : ["ITT","Game"]

Threads: ["Rate my Chickens", "ITT we are technicans", "Game of Thrones"]

Then only "ITT we are technicans" and "Game of Thrones" are gonna be saved.

Is this possible or already implented?

The reason is, on specific boards they are trolls which are posting the same garbage over and over again and i would like to ignore these automaticilly.

Greetings
Raymond

**Comment by diantahoc** [comment]
Yes this is already implemented using Thread Filters.

To use it, add a thread filter for board X and monitor board X in partial monitoring mode.

For example if I wanted to archive all threads from /3/ that contains `ITT We make 3d models` in the OP post, first add a filter like in the picture below:

![fitler](https://cloud.githubusercontent.com/assets/2142305/21265048/28b18f88-c3a8-11e6-9dd1-dd88759ce600.png)

and then monitor /3/ in Partial Monitoring mode:

![mon](https://cloud.githubusercontent.com/assets/2142305/21265074/46bd2096-c3a8-11e6-871e-ed3674744761.png)

ChanArchiver would then save all threads with the text `ITT We make 3d models`.

I am glad that you like this software.

**Comment by RaymondSchnyder** [comment]
I think i explained it wrong, i meant a filter, which excludes all These shit threads i dont want to see (rate my cock, furry,etc...) 
Is there a possibilty to exclude specific threads?

I want all threads but i want to exclude some threads 

sorry for missexplaining :/

**Comment by diantahoc** [comment]
What you want is a black list. I will implement that next week, when I have the time.

**Comment by hexafluoride** [comment]
Can't you just use negative lookaround? I don't know much regex, but I saw this on the internet:

    ^(?!foo$)  will match anything not equal to "foo"

https://forums.asp.net/t/1324649.aspx?Inverse+negated+regex+expression

**Comment by diantahoc** [comment]
@hexafluoride Hi hex, long time no see.

I think negative look around might work but a black list feature is way simpler when it comes to managing filters.

**Comment by RaymondSchnyder** [comment]
any update on this?

**Comment by diantahoc** [comment]
@RaymondSchnyder Yes.

I almost completely forgot about it, but I finally got a bit of spare time and I added `Blacklist Monitoring` mode, which I believe is what you want.

In `Blacklist Monitoring` mode, all threads are saved except those matching a `Blacklist` fitler. Threads matching a `Blacklist` filter are not saved.

To use it, just add filters matching troll threads to the black list:

![new page](https://cloud.githubusercontent.com/assets/2142305/21871329/28e197a6-d86a-11e6-9aee-f1cea3bbbfb0.png)

And then run your favorite board in `Blacklist Monitoring` mode:

![black list entry](https://cloud.githubusercontent.com/assets/2142305/21871337/39094430-d86a-11e6-9f38-c6bbff06c054.png)

It will now save all threads except troll threads.

Download this release from here:

https://github.com/diantahoc/chan-archiver/releases/tag/1.13

Regards,
